### DISCUSSION TITLE
put foam folder in different server not the same with ampache

### DISCUSSION THREAD

**Discussion by daoudi5** [Andrew-McGee/foam#12] [Q&A]
Hi @Andrew-McGee, Thank you so much for the great working you are doing and sharing it.
I want to know if possible to put foam folder in different location. For example: foam folder in server A with www.domainNameA.com
and Ampache server in another server B with www.domaineNameB.com.
I have tried but i can not find my tracks and album.

Thanks for you help

**Comment by Andrew-McGee** [comment]
Hi,

Yes I don't think it should be a problem. I have them on the same server so I will test later.  When you login to foam for the first time make sure you put the correct host name for your amapche server and the port. Like this: 
http://192.168.1.100:81

It shouldn't matter if you use IP address or hostname. Let me know if it works for you.

**Reply by daoudi5** [reply]
Hi,
Thank you for your answer. 
on my side it's not working.
This is the link with the same server : http://chant.note4research.com/foam/index.php
In this link it's working fine
But when I put it on another server using the host:port it not working
this is the link where it's not working : http://music.intmissioncenter.org/foam/index.php

My ampache server is : chant.note4research.com

I have this error : Warning: count(): Parameter must be an array or an object that implements Countable in /home/clients/f164082f1f8cfaddb97ed47e0c8ce8d8/sites/music.intmissioncenter.org/foam/albums_view.php on line 35

You can test with the visitor credentials.

Thank you for your help.
Best Regards

**Reply by Andrew-McGee** [reply]
Hi,
I did some tests. It's defintiely not making the connection which is why you get a php error. You get the same thing if you log in with incorrect credentials (my error checking is not working well enough to kick you out so you just see a blank page). The reason you see the error msg is due to a php setting. I think one server has it turned off and one turned on in the php settings. 

I have a test page here: http://foamdemo.mcgee.technology and I tried connecting to your ampache from there but it didn't work. I can connect to my remote Ampache from there and it works fine.

 Then I tried connecting to my Ampache from both of your servers and they both didn't work. I have a feeling maybe there is some firewall or network setting that is blocking either outbound or inbound connections on one or both servers. So it seems to allow a connection to itself internally but not one from outside over the internet. That is my guess so far. 

Can you connect to your normal Ampache login screen from an external machine? That would be a test if connection can be made. And I assume your Ampache server is on port 80? I have my Ampache server on port 81 and my Apache webserver on port 80. I wonder if ports could be an issue.
